compiler/hash-pest-parser/src/backend.rs

```diff
@@ -0,0 +1,66 @@
+//! Hash language parser backend implementation using pest
+//
+// All rights reserved 2021 (c) The Hash Language authors
+
+use crate::{
+    error::PestError,
+    grammar::{Grammar, Rule},
+    translate::PestAstBuilder,
+};
+use hash_ast::{
+    ast,
+    error::{ParseError, ParseResult},
+    parse::ParserBackend,
+    resolve::ModuleResolver,
+};
+use hash_utils::timed;
+use pest::Parser;
+use std::path::Path;
+
+pub struct PestBackend;
+
+impl ParserBackend for PestBackend {
```

Comment:
I am having to do the same pattern here for within `hash-parser` on the [tokeniser](https://github.com/hash-org/lang/tree/tokeniser) branch. Is there a way of getting rid of the `impl ParserBackend for ...` as it seems to be redundant.

Reply:
What do you mean? Which part is redundant?

Reply:
Since `PestBackend` and `HashBackend` carry no data on them, is there some way to just have `impl ParserBackend` for the crate. Then, by the magic of compile-time feature flags, we can just get the relevant crate and get the implementation we want?

Reply:
Well it doesn't work like that. Traits have to be implemented by types, there are no "free traits" like there are free functions. So you need a unit struct to implement the trait. This isn't considered an anti-pattern, and is used pretty commonly AFAIK.

Reply:
That's fair enough, I guess the unit struct won't even mean anything at runtime.